x-pack/plugins/security/server/spaces/secure_spaces_client_wrapper.ts

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from '@hapi/boom';
+import { KibanaRequest } from 'src/core/server';
+import { GetAllSpacesPurpose, GetSpaceResult } from '../../../spaces/common/model/types';
+import { Space, ISpacesClient } from '../../../spaces/server';
+import { LegacySpacesAuditLogger } from './legacy_audit_logger';
+import { SecurityPluginSetup } from '..';
+
+const PURPOSE_PRIVILEGE_MAP: Record<
+  GetAllSpacesPurpose,
+  (authorization: SecurityPluginSetup['authz']) => string[]
+> = {
+  any: (authorization) => [authorization.actions.login],
+  copySavedObjectsIntoSpace: (authorization) => [
+    authorization.actions.ui.get('savedObjectsManagement', 'copyIntoSpace'),
+  ],
+  findSavedObjects: (authorization) => {
+    return [authorization.actions.login, authorization.actions.savedObject.get('config', 'find')];
+  },
+  shareSavedObjectsIntoSpace: (authorization) => [
+    authorization.actions.ui.get('savedObjectsManagement', 'shareIntoSpace'),
+  ],
+};
+
+interface GetAllSpacesOptions {
+  purpose?: GetAllSpacesPurpose;
+  includeAuthorizedPurposes?: boolean;
+}
+
+export class SecureSpacesClientWrapper implements ISpacesClient {
+  private useRbac: boolean;
+
+  constructor(
+    private readonly spacesClient: ISpacesClient,
+    private readonly request: KibanaRequest,
+    private readonly authorization: SecurityPluginSetup['authz'],
+    private readonly legacyAuditLogger: LegacySpacesAuditLogger
+  ) {
+    this.useRbac = authorization.mode.useRbacForRequest(request);
+  }
+
+  public async getAll({
+    purpose = 'any',
+    includeAuthorizedPurposes,
+  }: GetAllSpacesOptions = {}): Promise<Space[]> {
```

Comment:
__question:__ do we intentionally cast `GetSpaceResult[]` to `Space[]`?

Reply:
Ah good catch - I think this should be `GetSpaceResult[]` instead.